front/pages/api/w/[wId]/assistant/builder/process/generate_schema.ts

```diff
@@ -0,0 +1,127 @@
+import type {
+  ProcessSchemaPropertyType,
+  WithAPIErrorReponse,
+} from "@dust-tt/types";
+import { cloneBaseConfig } from "@dust-tt/types";
+import { InternalPostBuilderProcessActionGenerateSchemaRequestBodySchema } from "@dust-tt/types";
+import { DustProdActionRegistry } from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { runAction } from "@app/lib/actions/server";
+import { Authenticator, getSession } from "@app/lib/auth";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorReponse<{
+      schema: ProcessSchemaPropertyType[];
+    }>
+  >
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner || !auth.isUser()) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "app_auth_error",
+        message:
+          "Workspace not found or user not authenticated to this workspace.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "POST":
+      const bodyValidation =
+        InternalPostBuilderProcessActionGenerateSchemaRequestBodySchema.decode(
+          req.body
+        );
+
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
```

Comment:
we have [this](https://github.com/dust-tt/dust/blob/4295a6c4e5c3563e189fecd32bc13655f93a3ce0/types/src/shared/utils/iots_utils.ts#L47-L57) now

Reply:
Not much shorter but switching to it :)

![Screenshot from 2024-05-16 17-36-17](https://github.com/dust-tt/dust/assets/15067/ea773140-2f39-4f02-842e-693d715a111e)

Reply:
True it's not that helpful :p

Reply:
slightly more readable I guess

Reply:
def